climada/util/yearsets.py

```diff
@@ -0,0 +1,266 @@
+        idx += nr_events[year]
+
+    return impact_per_year, nr_events, sampling_vect
+
+
+def sampling_uniform(tot_nr_events, nr_input_events):
+    """Sample uniformely from an array (nr_input_events) for a given amount of
+    events (number events)
+
+    INPUT:
+        tot_nr_events (int): number of events to be resampled
+        nr_input_events (int): number of events contained in given event impact set (eis)
+
+    OUTPUT:
+        sampling_vect (array): sampling vector
+      """
+
+    repetitions = np.ceil(tot_nr_events/(nr_input_events-1)).astype('int')
+
+    rng = default_rng()
+    if repetitions >= 2:
+        sampling_vect = np.round(rng.choice((nr_input_events-1)*repetitions,
+                                            size=tot_nr_events, replace=False)/repetitions
+                                 ).astype('int')
+    else:
+        sampling_vect = rng.choice((nr_input_events-1), size=tot_nr_events,
+                                   replace=False).astype('int')
+
+    return sampling_vect
+
+def sampling_poisson(nr_resampled_years, nr_annual_events, nr_input_events):
+    """Sample amount of events per year following a Poisson distribution
+
+    INPUT:
+        nr_resampled_years (int): the target number of years the impact yearset shall
+            contain.
+        nr_annual_events (int): number of events per year in given event impact set
+        nr_input_events (int): number of events contained in given event impact set (eis)
+
+    OUTPUT:
+        sampling_vect (array): sampling vector
+        nr_events_per_year (array): number of events per resampled year
+    """
+
+    nr_events_per_year = np.round(np.random.poisson(lam=nr_annual_events,
+                                                    size=nr_resampled_years)).astype('int')
+
+    #non_zero_years =  np.where(nr_events_per_year != 0)
+    tot_nr_events = sum(nr_events_per_year)
+
+    sampling_vect = sampling_uniform(tot_nr_events, nr_input_events)
+
+    return sampling_vect, nr_events_per_year
+
+def calculate_correction_fac(impact_per_year, nr_resampled_years, eis):
+    """Apply a correction factor to ensure the expected annual impact (eai) of the annual
+    impact set(ais) amounts to the eai of the event impact set (eis)
+
+    INPUT:
+        impact_per_year (array): resampled annual impact set before applying the correction factor
+        nr_resampled_years (int): the target number of years the annual impact set contains
+        eis (impact class object): event impact set
+
+    OUTPUT:
+        correction_factor (int): the correction factor is calculated as eis_eai/ais_eai
+    """
+
+    ais_eai = np.sum(impact_per_year)/nr_resampled_years
+    eis_eai = np.sum(eis.frequency*eis.at_event)
+    correction_factor = eis_eai/ais_eai
+    LOGGER.info("The correction factor amounts to %s", (correction_factor-1)*100)
+
+    return correction_factor
+
+def wrapper_multi_impact(list_impacts, nr_resampled_years):
```

Comment:
Is this function really needed? It is a one-liner ...

Reply:
maybe not, I just started writing it as I was thinking about multi-hazard. I could imagine it will get more complex once these thoughts are fleshed out...

Reply:
Interresting! What do you mean it will get more complex?